x-pack/plugins/apm/server/lib/services/get_services/get_service_transaction_stats.ts

```diff
@@ -0,0 +1,189 @@
+                ...metrics,
+                environments: {
+                  terms: {
+                    field: SERVICE_ENVIRONMENT,
+                    missing: '',
+                  },
+                },
+                agentName: {
+                  top_hits: {
+                    docvalue_fields: [AGENT_NAME] as const,
+                    size: 1,
+                  },
+                },
+                timeseries: {
+                  date_histogram: {
+                    field: '@timestamp',
+                    fixed_interval: getBucketSize({
+                      start,
+                      end,
+                      numBuckets: 20,
+                    }).intervalString,
+                    min_doc_count: 0,
+                    extended_bounds: { min: start, max: end },
+                  },
+                  aggs: metrics,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const deltaAsMinutes = (setup.end - setup.start) / 1000 / 60;
+
+  return (
+    response.aggregations?.services.buckets.map((bucket) => {
+      const topTransactionTypeBucket =
+        bucket.transactionType.buckets.find(
+          ({ key }) =>
+            key === TRANSACTION_REQUEST || key === TRANSACTION_PAGE_LOAD
+        ) ?? bucket.transactionType.buckets[0];
+
+      return {
+        serviceName: bucket.key as string,
+        transactionType: topTransactionTypeBucket.key as string,
+        environments: topTransactionTypeBucket.environments.buckets
+          .map((environmentBucket) => environmentBucket.key as string)
+          .filter(Boolean),
+        agentName: topTransactionTypeBucket.agentName.hits.hits[0].fields[
+          'agent.name'
+        ]?.[0] as AgentName,
+        avgResponseTime: {
+          value: topTransactionTypeBucket.avg_duration.value,
+          timeseries: topTransactionTypeBucket.timeseries.buckets.map(
+            (dateBucket) => ({
+              x: dateBucket.key,
+              y: dateBucket.avg_duration.value,
+            })
+          ),
+        },
+        transactionErrorRate: {
+          value:
+            topTransactionTypeBucket.failure.count.value /
+            (topTransactionTypeBucket.failure.count.value +
+              topTransactionTypeBucket.successful.count.value),
+          timeseries: topTransactionTypeBucket.timeseries.buckets.map(
+            (dateBucket) => ({
+              x: dateBucket.key,
+              y:
+                dateBucket.failure.count.value /
+                (dateBucket.failure.count.value +
+                  dateBucket.successful.count.value),
+            })
+          ),
+        },
+        transactionsPerMinute: {
+          value: topTransactionTypeBucket.count.value / deltaAsMinutes,
+          timeseries: topTransactionTypeBucket.timeseries.buckets.map(
+            (dateBucket) => ({
+              x: dateBucket.key,
+              y: dateBucket.avg_duration.value,
```

Comment:
This is avg duration as well, should be `dateBucket.count.value / deltaAsMinutes`